.dialyzer.ignore-warnings

```diff
@@ -45,6 +34,8 @@ apps/evm/lib/evm/gas.ex:309: Function operation_cost/3 has no local return
 apps/evm/lib/evm/gas.ex:309: The call 'Elixir.EVM.Gas':operation_cost(__@1::any(),__@2::any(),__@3::any(),'nil') breaks the contract (atom(),['Elixir.EVM':val()],['Elixir.EVM':val()],'Elixir.EVM.MachineState':t()) -> t() | 'nil'
 apps/evm/lib/evm/gas.ex:576: Function gas_cost_for_nested_operation/2 will never be called
 apps/evm/lib/evm/machine_state.ex:53: Function subtract_gas/2 has no local return
+apps/evm/lib/evm/machine_state.ex:55: The pattern {'changed', _cost@1, _new_call_gas@1} can never match the type number() | {'original',number()}
```

Comment:
Could this possibly be a real error? I'm not sure what exposed this but it does look like the spec for Gas.cost would never match this case?
```ex
# apps/evm/lib/evm/gas.ex:136
  @spec cost(MachineState.t(), ExecEnv.t(), keyword()) :: t | nil | {atom(), t | nil}

# apps/evm/lib/evm/machine_state.ex:54
    case Gas.cost(machine_state, exec_env, with_status: true) do
      {:changed, cost, new_call_gas} ->

```

Reply:
Yeah, I think it makes sense to look into these and see if they're real errors

Reply:
After further research, I think we should look at these errors, but not in relation to this PR. I think they are unrelated to this. I just improved a typespec that surfaced these errors.